lib/sqlitex.ex

```diff
@@ -7,19 +7,47 @@ defmodule Sqlitex do
     :esqlite3.open(path)
   end
 
-  def query(db, query, opts \\ []) do
-    {:ok, statement} = :esqlite3.prepare(query, db)
-    types = :esqlite3.column_types(statement) |> Tuple.to_list
-    columns = :esqlite3.column_names(statement) |> Tuple.to_list
-    rows = :esqlite3.fetchall(statement)
-    into = Keyword.get(opts, :into, [])
-    Sqlitex.Row.from(types, columns, rows, into)
-  end
-
   def with_db(path, fun) do
     {:ok, db} = open(path)
     res = fun.(db)
     close(db)
     res
   end
+
+  def exec(db, sql) do
+    :esqlite3.exec(sql, db)
+  end
+
+  def query(db, sql, opts \\ []) do
+    {params, into} = query_options(opts)
+    {:ok, statement} = :esqlite3.prepare(sql, db)
+    :ok = :esqlite3.bind(statement, params)
+    types = :esqlite3.column_types(statement)
+    columns = :esqlite3.column_names(statement)
+    rows = :esqlite3.fetchall(statement)
+    return_rows_or_error(types, columns, rows, into)
+  end
+
+  defp query_options(opts) do
+    params = Keyword.get(opts, :bind, [])
+    into = Keyword.get(opts, :into, [])
+    {params, into}
+  end
+
+  defp return_rows_or_error({:error, kind} = error, columns, rows, into) do
+    case kind do
+      :no_columns -> return_rows_or_error({}, columns, rows, into)
+      _ -> error
+    end
+  end
+  defp return_rows_or_error(types, {:error, kind} = error, rows, into) do
+    case kind do
+      :no_columns -> return_rows_or_error(types, {}, rows, into)
+      _ -> error
+    end
+  end
```

Comment:
I knew once I started working with Ecto I would encounter this problem again.  The reason this needs to ignore the :no_columns errors is because Ecto universally uses `query()` to issue writable as well as readable commands to the database -- which is perfectly valid and allowed by the SQLite3 API.  However, for, e.g., the CREATE command, one doesn't expect column types or names so even though `sqlite3_column_count()` returns 0, this is expected, and the error is unnecessary.